src/presentation/http/http-api.ts

```diff
@@ -0,0 +1,236 @@
+import { getLogger } from '@infrastructure/logging/index.js';
+import type { HttpApiConfig } from '@infrastructure/config/index.js';
+import type { FastifyInstance } from 'fastify';
+import type { FastifyBaseLogger } from 'fastify';
+import fastify from 'fastify';
+import type Api from '@presentation/api.interface.js';
+import type { DomainServices } from '@domain/index.js';
+import cors from '@fastify/cors';
+import fastifyOauth2 from '@fastify/oauth2';
+import fastifySwagger from '@fastify/swagger';
+import fastifySwaggerUI from '@fastify/swagger-ui';
+import initMiddlewares, { type Middlewares } from '@presentation/http/middlewares/index.js';
+import cookie from '@fastify/cookie';
+import NotFoundDecorator from './decorators/notFound.js';
+import NoteRouter from '@presentation/http/router/note.js';
+import OauthRouter from '@presentation/http/router/oauth.js';
+import AuthRouter from '@presentation/http/router/auth.js';
+import UserRouter from '@presentation/http/router/user.js';
+import AIRouter from '@presentation/http/router/ai.js';
+import EditorToolsRouter from './router/editorTools.js';
+import { UserSchema } from './schema/User.js';
+import type { RequestParams, Response } from '@presentation/api.interface.js';
+
+
+const appServerLogger = getLogger('appServer');
+
+/**
+ * Http server implementation
+ */
+export default class HttpApi implements Api {
+  /**
+   * Fastify server instance
+   */
+  private server: FastifyInstance | undefined;
+
+  /**
+   * Http server config
+   */
+  private config: HttpApiConfig | undefined;
+
+  /**
+   * Initializes http server
+   *
+   * @param config - http server config
+   * @param domainServices - instances of domain services
+   */
+  public async init(
+    config: HttpApiConfig,
+    domainServices: DomainServices
+  ): Promise<void> {
+    this.server = fastify({
+      logger: appServerLogger as FastifyBaseLogger,
+    });
+    this.config = config;
```

Comment:
I'd suggest passing config through the constructor, so you don't need to pass it to every method below

Reply:
Better to leave it as it is now. Otherwise we'll have to add these lines into each method
```
if (this.config === undefined) {
      return;
}
```

Reply:
fixed